src/java.base/share/classes/java/net/ServerSocket.java

```diff
@@ -711,12 +709,18 @@ private void ensureCompatible(SocketImpl si) throws IOException {
      * @revised 1.4
      */
     public void close() throws IOException {
-        synchronized(closeLock) {
-            if (isClosed())
-                return;
-            if (created)
-                impl.close();
-            closed = true;
+        synchronized (stateLock) {
```

Comment:
Maybe makes some sense to check `closed` outside the lock here? Looks like we don't need to re-acquire the lock for repeated `close()` calls.

Reply:
That would be okay too although I wouldn't expect it is very common to attempt to close a ServerSocket more than once (the only scenario that comes to mind is where there is an async close which triggers a shutdown to server and the shutdown invokes close too).